express-notes-api/server.js

```diff
@@ -0,0 +1,184 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+/* This just makes it easier to create error messages for me.
+   And it looks cleaner when written in the code later on. */
+class Error {
+  constructor(errorMsg) {
+    this.error = errorMsg;
+  }
+}
+
+const port = 8080;
+const readFile = './data.json';
+const writeFile = './data.json';
+const app = express();
+const parser = express.json();
+app.use(parser);
+
+/* An async function for grabbing the data from the json file.
+   Data will be parsed before returning.                    */
+async function getJSON(fileUrl) {
+  try {
+    const contents = await fs.readFile(fileUrl, 'utf8');
+    const parsed = JSON.parse(contents);
+    return parsed;
+  } catch (err) {
+    console.log('Error reading json:', err);
+  }
+}
+
+/* An async function for writing our updated data back to the json file. */
+async function writeJSON(fileUrl, data) {
```

Comment:
This one also would be nicer if it did not take `fileUrl` as a parameter, but just had one parameter, `data`.